underfs/hdfs/src/main/java/alluxio/underfs/hdfs/activesync/SupportedHdfsActiveSyncProvider.java

```diff
@@ -258,38 +273,52 @@ public void stopSync(AlluxioURI ufsUri) {
    * @param eventStream event stream
    */
   public void pollEvent(DFSInotifyEventInputStream eventStream) {
-    EventBatch batch;
     LOG.debug("Polling thread starting, with timeout {} ms", mActiveUfsPollTimeoutMs);
-    int count = 0;
     long start = System.currentTimeMillis();
 
     long behind = eventStream.getTxidsBehindEstimate();
 
     while (!Thread.currentThread().isInterrupted()) {
       try {
-        batch = eventStream.poll(mActiveUfsPollTimeoutMs, TimeUnit.MILLISECONDS);
 
-        if (batch != null) {
-          long txId = batch.getTxid();
-          count += batch.getEvents().length;
-          for (Event event : batch.getEvents()) {
-            processEvent(event, mUfsUriList, txId);
+        List<Callable<Integer>> process = new LinkedList<>();
+        for (int i = 0; i < mBatchSize; i++) {
+          EventBatch batch = eventStream.poll(mActiveUfsPollTimeoutMs, TimeUnit.MILLISECONDS);
+          if (batch == null) {
+            break;
           }
+          EventBatch finalBatch = batch;
```

Comment:
Is this supposed to be `final`? Can the previous `EventBatch batch = ...` just be `final`?

Reply:
updated